src/cascadia/TerminalSettingsEditor/ActionsViewModel.cpp

```diff
@@ -18,383 +24,1194 @@ using namespace winrt::Windows::UI::Xaml::Data;
+                if (wrapper)
+                {
+                    weak->PropagateWindowRootRequested.raise(*weak, wrapper);
+                }
             }
         });
+        if (_IsNewCommand)
+        {
+            // for new commands, make sure we generate a new ID everytime any arg value changes
+            actionArgsVM.WrapperValueChanged([weakThis = get_weak()](const IInspectable& /*sender*/, const IInspectable& /*args*/) {
+                if (auto weak = weakThis.get())
+                {
+                    weak->_command.GenerateID();
+                }
+            });
+        }
+        else if (!IsUserAction())
+        {
+            actionArgsVM.WrapperValueChanged([weakThis = get_weak()](const IInspectable& /*sender*/, const IInspectable& /*args*/) {
+                if (auto weak = weakThis.get())
+                {
+                    weak->_ReplaceCommandWithUserCopy(false);
+                }
+            });
+        }
+    }
+
+    void CommandViewModel::_ReplaceCommandWithUserCopy(bool reinitialize)
+    {
+        // the user is attempting to edit an in-box action
+        // to handle this, we create a new command with the new values that has the same ID as the in-box action
+        // swap out our underlying command with the copy, tell the ActionsVM that the copy needs to be added to the action map
+        if (const auto actionsPageVM{ _actionsPageVM.get() })
+        {
+            const auto newCmd = Model::Command::CopyAsUserCommand(_command);
+            _command = newCmd;
+            actionsPageVM.AttemptAddCopiedCommand(_command);
+            if (reinitialize)
+            {
+                // full reinitialize needed, recreate the action args VM
+                // (this happens when the shortcut action is being changed on an in-box action)
+                _CreateAndInitializeActionArgsVMHelper();
+            }
+            else
+            {
+                // no need to reinitialize, just swap out the underlying data model
+                // (this happens when an additional argument is being changed on an in-box action)
+                auto actionArgsVMImpl{ get_self<ActionArgsViewModel>(_ActionArgsVM) };
+                actionArgsVMImpl->ReplaceActionAndArgs(_command.ActionAndArgs());
+            }
+        }
     }
 
-    hstring KeyBindingViewModel::EditButtonName() const noexcept { return RS_(L"Actions_EditButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
-    hstring KeyBindingViewModel::CancelButtonName() const noexcept { return RS_(L"Actions_CancelButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
-    hstring KeyBindingViewModel::AcceptButtonName() const noexcept { return RS_(L"Actions_AcceptButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
-    hstring KeyBindingViewModel::DeleteButtonName() const noexcept { return RS_(L"Actions_DeleteButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }

```

Comment:
Huh, the delete button doesn't have a tooltip.

Also, I might've mentioned it before, these need the accessibility names still.